x-pack/plugins/security_solution/server/lib/detection_engine/signals/threat_mapping/get_threat_list.ts

```diff
@@ -58,15 +55,17 @@ export const getThreatList = async ({
       ...threatListConfig,
       query: queryFilter,
       search_after: searchAfter,
-      sort: ['_doc', { '@timestamp': 'asc' }],
+      sort: ['_shard_doc', { '@timestamp': 'asc' }],
     },
     track_total_hits: false,
-    ignore_unavailable: true,
-    index,
-    size: calculatedPerPage,
+    size: INDICATOR_PER_PAGE,
```

Comment:
Using a static value here will break the `concurrentSearches` and `itemsPerSearch` functionality provided through rule params. Are we planning to remove those params?

Reply:
this is a great question. I believe that concurrent search logic has been built into indicator match rules a while back with the intention of speeding up the rule execution. That being said, there seems to be no test coverage around the feature, and as far as I am aware the parameters `perPage` and `concurrentSearches` are undocumented and can only be set through the API - which makes me think that anyone creating indicator match rules via the UI have never been able to use anything but the default values: 1 for `concurrentSearches` and 9000 for `itemsPerSearch`. I am under the impression that the feature of executing concurrent searches is not something we have ever intentionally released - some telemetry data could confirm this. Is it possible to see how many rule SOs actually have "itemsPerSearch" and "concurrentSearches" values set out there in the world?

In my opinion it makes sense to be intentional about the concurrent searching capability of Indicator Match rules by bringing it into 8.3 planning as a part of the performance epic and reevaluate the performance with our improved benchmarking tooling. Until then, we should consider the feature unsupported - I'll get some product input on this.

@peluja1012 @jethr0null @shimonmodi

Reply:
With the latest commit I brought back `concurrentSearches`, `itemsPerSearch`, the `perPage` variable to ensure that the built-in concurrent searching logic remains intact. Created https://github.com/elastic/security-team/issues/3465 to benchmark concurrent threat searches for 8.3